numpy/core/tests/test_shape_base.py

```diff
@@ -224,13 +224,26 @@ def test_exceptions(self):
         assert_raises(ValueError, concatenate, (0,))
         assert_raises(ValueError, concatenate, (np.array(0),))
 
+        # dimensionality must match
+        assert_raises_regex(
+            ValueError,
+            r"all the input arrays must have same number of dimensions, but "
+            r"the array at index 0 has 1 dimension\(s\) and the array at "
+            r"index 1 has 2 dimension\(s\)",
+            np.concatenate, (np.zeros(1), np.zeros((1, 1))))
+
         # test shapes must match except for concatenation axis
         a = np.ones((1, 2, 3))
         b = np.ones((2, 2, 3))
         axis = list(range(3))
         for i in range(3):
             np.concatenate((a, b), axis=axis[0])  # OK
-            assert_raises(ValueError, np.concatenate, (a, b), axis=axis[1])
+            assert_raises_regex(
+                ValueError,
+                "all the input array dimensions for the concatenation axis "
+                "must match exactly, but along dimension 1, the array at "
+                "index 0 has size 1 and the array at index 1 has size 2",
+                np.concatenate, (a, b), axis=axis[1])
```

Comment:
The produced message has `along dimension 0`

Reply:
Now fixed (the single test failure is unrelated).

As a side note, the test is written in a rather convoluted manner, as the loop over `i` only matters because `a` and `b` are being replaced by moveaxis'ed versions of themselves below :/

Reply:
Why is it unrelated? The azure test failure is in this test.

Reply:
Oh, sorry, was looking at the travis failure.

Reply:
Something is overflowing, the azure failure message ends with "but the array at index 0 has 1 dimension(s) and the array at index 1 has 140733193388034 dimension(s)"

Reply:
Looks like I need to use %d instead of %zd.